python/cudf/cudf/core/frame.py

```diff
@@ -2069,7 +2069,12 @@ def from_arrow(cls, data):
                 else:
                     dtype = np_dtypes[name]
 
-                result._data[name] = result._data[name].astype(dtype)
+                if cudf.utils.dtypes.is_struct_dtype(dtype):
+                    result._data[name] = result._data[name]._rename_fields(
+                        [field.name for field in data[name].type]
+                    )
+                elif not cudf.utils.dtypes.is_list_dtype(dtype):
```

Comment:
Should this be just else, or we should add an else to raise an error ?

Reply:
Changed the if/elif structure a little bit to make things even clear. This is just to handle some special cases and actual dtypes are already loaded, only in these special cases we either rename fields(incase of struct) or type-cast(categorical, boolean, object). So I don't think throwing error here might not be needed as that would indicate it's a valid type.